### ISSUE TITLE
Mongoose "Map" field mapping

### ISSUE THREAD

**Issue by MatthewJonD** [DScheglov/mongoose-schema-jsonschema#26]
I have a Mongoose schema defined with the "map" field type.  Looks like-
```
    let richMetaDataItemSchema = mongoose.Schema({
        name: { type: String },
        language: {
            type: Map,
            of: { type: String }
        }
    });
```

Right now its getting transformed into a schema that looks like-
```
 "summary": {
      "title": "summary",
      "type": "object",
      "properties": {
        "name": {
          "type": "string"
        },
        "language": {
          "type": "map",
          "properties": {
            "$*": {
              "type": "string"
            }
          }
        }
      },
```
Problem is the type: "map" isnt a valid type(Or not that I can find in the docs).
I am also using the tool-https://uniforms.tools/
And that also has a problem with that field type...

Any ideas, or possible to support a valid JSONSchema construct?

**Comment by DScheglov** [comment]
Hey, @MatthewJonD I will tackle the issue, but I'm a little bit surprised that mongoose supports `Map`.
In general such cases are considered to be covered with custom types.

```js
class MongooseMap extends Map {
  jsonSchema() {
    return { type: "object", properties: { ... } }; // or { type: "array", item: { type: "array", item: [{type: "string" }, { type: string }] }
  }
}
```

So, If Map is often used I will add support directly for Map.
But currently I'm going to implement Custom type: `MongooseMap` that supports jsonSchema.

**Comment by DScheglov** [comment]
@MatthewJonD , fix in version `v1.4.5` by adding support for type `Map`.
So, you don't need to change your code.

Feel free to reopen the issue if something works wrong.
